app/services/ExactTargetService.scala

```diff
@@ -168,32 +168,39 @@ class ExactTargetService(
       daysAllowed: Int
   )(implicit zuoraRestService: ZuoraRestService[Future]): Future[Unit] = {
 
-    val buildDataExtensionRow =
-      EitherT(GetSalesforceContactForSub(subscription)(zuoraService, salesforceService.repo, executionContext).flatMap { salesforceContact =>
-        EitherT(zuoraRestService.getAccount(subscription.accountId)).map { account =>
-          HolidaySuspensionBillingScheduleDataExtensionRow(
-            email = account.billToContact.email,
-            saluation = constructSalutation(salesforceContact.title, salesforceContact.firstName, Some(salesforceContact.lastName)),
-            subscriptionName = subscription.name.get,
-            subscriptionCurrency = subscription.currency,
-            packageName = packageName,
-            billingSchedule = billingSchedule,
-            numberOfSuspensionsLinedUp = numberOfSuspensionsLinedUp,
-            daysUsed = daysUsed,
-            daysAllowed = daysAllowed
-          )
-        }.run
-      })
+    def createDataExtensionRow(salesforceContact: Contact, zuoraAccount: ZuoraRestService.AccountSummary) = {
+      HolidaySuspensionBillingScheduleDataExtensionRow(
+        email = zuoraAccount.billToContact.email,
+        saluation = constructSalutation(salesforceContact.title, salesforceContact.firstName, Some(salesforceContact.lastName)),
+        subscriptionName = subscription.name.get,
+        subscriptionCurrency = subscription.currency,
+        packageName = packageName,
+        billingSchedule = billingSchedule,
+        numberOfSuspensionsLinedUp = numberOfSuspensionsLinedUp,
+        daysUsed = daysUsed,
+        daysAllowed = daysAllowed
+      )
+    }
 
-    buildDataExtensionRow.flatMap { row =>
-      EitherT(SqsClient.sendDataExtensionToQueue(Config.holidaySuspensionEmailQueue, row).map {
+    def sendToQueue(salesforceContact: Contact, row: HolidaySuspensionBillingScheduleDataExtensionRow) = {
+      SqsClient.sendDataExtensionToQueue(Config.holidaySuspensionEmailQueue, row, salesforceContact.identityId, salesforceContact.salesforceContactId).map {
         case Success(sendMsgResult) => \/.right(())
         case Failure(e) => \/.left(s"Details were: ${row.subscriptionName}, ${e.toString}")
-      })
-    }.run.map {
+      }
+    }
+
+    val enqueueResult = for {
+      salesforceContact <- EitherT.right(GetSalesforceContactForSub(subscription)(zuoraService, salesforceService.repo, executionContext))
```

Comment:
We need to check that this doesn't require an Identity ID anywhere in the lookup, as holiday suspension emails can go to anyone.

Reply:
You mean for the GetSalesforceContactForSub call? This functionality shouldn't have changed from before